src/bit-systems/particle-emitter.ts

```diff
@@ -0,0 +1,61 @@
+import { defineQuery, enterQuery, exitQuery } from "bitecs";
+import { ParticleEmitter } from "lib-hubs/packages/three-particle-emitter/lib/esm/index";
+import { Texture } from "three";
+
+import { HubsWorld } from "../app";
+import { ParticleEmitterTag } from "../bit-components";
+import { JobRunner } from "../utils/coroutine-utils";
+import { proxiedUrlFor } from "../utils/media-url-utils";
+import { resolveUrl, textureLoader } from "../utils/media-utils";
+
+function* setTexture(world: HubsWorld, eid: number) {
+  let src = APP.sid2str.get(ParticleEmitterTag.src[eid]);
+
+  const result: URL = yield resolveUrl(src);
+  let canonicalUrl = result.origin;
+
+  // handle protocol relative urls
+  if (canonicalUrl.startsWith("//")) {
+    canonicalUrl = location.protocol + canonicalUrl;
+  }
+
+  // todo: we don't need to proxy for many things if the canonical URL has permissive CORS headers
+  src = proxiedUrlFor(canonicalUrl);
+
+  const texture: Texture = yield textureLoader.loadAsync(src);
+
+  const particleEmitter: ParticleEmitter = world.eid2obj.get(eid)! as ParticleEmitter;
+  particleEmitter.material.uniforms.map.value = texture;
+  particleEmitter.visible = true;
+
+  ParticleEmitterTag.updateParticles[eid] = 1;
+}
+
+const particleEmitterQuery = defineQuery([ParticleEmitterTag]);
+const particleEmitterEnterQuery = enterQuery(particleEmitterQuery);
+const particleEmitterExitQuery = exitQuery(particleEmitterQuery);
+
+const jobs = new JobRunner();
+export function particleEmitterSystem(world: HubsWorld) {
+  particleEmitterEnterQuery(world).forEach(eid => {
+    jobs.stop(eid);
```

Comment:
We only ever add job for an entity in the enter query, and always stop it in the exit query so I don't think its possible for there to be a job for this entity already. At first reading this was confusing because I went looking for a way this might happen.